operations/mimir-mixin/dashboards/reads.libsonnet

```diff
@@ -212,40 +212,80 @@ local filename = 'mimir-reads.json';
         $.panel(title) +
         $.queryPanel(
           [
-            'kube_horizontalpodautoscaler_spec_min_replicas{%s, horizontalpodautoscaler="%s"}' % [$.namespaceMatcher(), $._config.autoscaling.querier.hpa_name],
-            'kube_horizontalpodautoscaler_spec_max_replicas{%s, horizontalpodautoscaler="%s"}' % [$.namespaceMatcher(), $._config.autoscaling.querier.hpa_name],
-            'kube_horizontalpodautoscaler_status_current_replicas{%s, horizontalpodautoscaler="%s"}' % [$.namespaceMatcher(), $._config.autoscaling.querier.hpa_name],
+            |||
+              kube_horizontalpodautoscaler_spec_max_replicas{%(namespace_matcher)s, horizontalpodautoscaler=~"%(hpa_name)s"}
+              # Add the scaletargetref_name label which is more readable than "kube-hpa-..."
+              + on (%(cluster_labels)s, horizontalpodautoscaler) group_left (scaletargetref_name)
+                0*kube_horizontalpodautoscaler_info{%(namespace_matcher)s, horizontalpodautoscaler=~"%(hpa_name)s"}
+            ||| % {
+              namespace_matcher: $.namespaceMatcher(),
+              cluster_labels: std.join(', ', $._config.cluster_labels),
+              hpa_name: $._config.autoscaling.querier.hpa_name,
+            },
+            |||
+              kube_horizontalpodautoscaler_status_current_replicas{%(namespace_matcher)s, horizontalpodautoscaler=~"%(hpa_name)s"}
+              # HPA doesn't go to 0 replicas, so we multiply by 0 if the HPA is not active.
+              * on (%(cluster_labels)s, horizontalpodautoscaler)
+                kube_horizontalpodautoscaler_status_condition{%(namespace_matcher)s, horizontalpodautoscaler=~"%(hpa_name)s", condition="ScalingActive", status="true"}
+              # Add the scaletargetref_name label which is more readable than "kube-hpa-..."
+              + on (%(cluster_labels)s, horizontalpodautoscaler) group_left (scaletargetref_name)
+                0*kube_horizontalpodautoscaler_info{%(namespace_matcher)s, horizontalpodautoscaler=~"%(hpa_name)s"}
+            ||| % {
+              namespace_matcher: $.namespaceMatcher(),
+              cluster_labels: std.join(', ', $._config.cluster_labels),
+              hpa_name: $._config.autoscaling.querier.hpa_name,
+            },
           ],
           [
-            'Min',
-            'Max',
-            'Current',
+            'Max {{ scaletargetref_name }}',
+            'Current {{ scaletargetref_name }}',
           ],
         ) +
         $.panelDescription(
           title,
           |||
-            The minimum, maximum, and current number of querier replicas.
+            The maximum, and current number of querier replicas.
+            Please note that the current number of replicas can still show 1 replica even when scaled to 0.
+            Since HPA never reports 0 replicas, the query will report zero only if the HPA is not active.
```

Comment:
```suggestion
            Since HPA never reports 0 replicas, the query will report 0 only if the HPA is not active.
```

Reply:
Done in https://github.com/grafana/mimir/pull/3962/commits/734ccb49ddaf7be0723bbaf9e6081e5ecbc4c999